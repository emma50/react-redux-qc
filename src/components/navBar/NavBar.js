import React from 'react';
import { useHistory, Link } from "react-router-dom";
import clsx from 'clsx';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import Divider from '@material-ui/core/Divider';
import Drawer from '@material-ui/core/Drawer';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import CssBaseline from '@material-ui/core/CssBaseline';
import List from '@material-ui/core/List';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/Button';
import Button from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';

const drawerWidth = 240;

const useStyles = makeStyles((theme) => ({
  root: {
    // overflowX: 'hidden',
    [theme.breakpoints.up('md')]: {
      display: 'none',
    },
  },
  appBar: {
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginRight: drawerWidth,
  },
  title: {
    flexGrow: 1,
    fontSize: 'small',
  },
  hide: {
    display: 'none',
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    padding: theme.spacing(0, 1),
    // necessary for content to be below app bar
    ...theme.mixins.toolbar,
    justifyContent: 'flex-start',
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    marginRight: -drawerWidth,
  },
  contentShift: {
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginRight: 0,
  },
  // Desktop view
  desktopRoot: {
    flexGrow: 1,
    // position: 'fixed',
    [theme.breakpoints.down('sm')]: {
      display: 'none',
    },
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  desktopTitle: {
    flexGrow: 1,
    fontSize: 'small',
  },
  fontSize: {
    fontSize: 'small',
  },
  link: {
    textDecoration: 'none',
    color: 'white',
  },
  // hiddenForMobile: {
  //   [theme.breakpoints.down('sm')]: {
  //     display: 'none',
  //   },
  // },
}));

export default function PersistentDrawerRight(props) {
  const history = useHistory()
  const classes = useStyles();
  const theme = useTheme();
  const [open, setOpen] = React.useState(false);

  const handleDrawerOpen = () => {
    setOpen(true);
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };

  return (
    <div>
      <div className={classes.root}>
        <CssBaseline />
        <AppBar
          position="fixed"
          className={clsx(classes.appBar, {
            [classes.appBarShift]: open,
          })}
        >
          <Toolbar>
            <Typography variant="h6" noWrap className={classes.title}>
              QUICK CREDIT
            </Typography>
            <IconButton
              color="inherit"
              aria-label="open drawer"
              edge="end"
              onClick={handleDrawerOpen}
              className={clsx(open && classes.hide)}
            >
              <MenuIcon />
            </IconButton>
          </Toolbar>
        </AppBar>
        <Drawer
          className={classes.drawer}
          variant="persistent"
          anchor="right"
          open={open}
          classes={{
            paper: classes.drawerPaper,
          }}
        >
          <div className={classes.drawerHeader}>
            <IconButton onClick={handleDrawerClose}>
              {theme.direction === 'rtl' ? <ChevronLeftIcon /> : <ChevronRightIcon />}
            </IconButton>
          </div>
          <Divider />
          <List>
            {[
              { text: 'Apply Now', url: '/applyforloan' }, 
              { text: 'Home', url: '/' },
              { text: 'Contact', url: '' },
              { text: 'Login', url: '/signin' },
              { text: 'Register', url: '/signup' },
             ].map((item, index) => (
              <Link to={item.url} key={index} style={{'color': 'black', 'textDecoration': 'none'}}>
                <ListItem button>
                  <ListItemText primary={item.text}/>
                </ListItem>
              </Link>
            ))}
          </List>
        </Drawer>
      </div>
      <div className={classes.desktopRoot}>
        <AppBar position="fixed" className={classes.hiddenForMobile}>
          <Toolbar>
            <Typography variant="h6" className={classes.desktopTitle}>
              <Button color="inherit" style={{'fontSize': 'small'}}>QUICK CREDIT</Button>
            </Typography>
            <Typography>
              <Button 
                //to='/applyforloan'
                color="inherit"
                // className={clsx(classes.desktopTitle, classes.link)}
                className={classes.desktopTitle}
                onClick={() => history.push('/applyforloan')} 
              > 
                Apply Now
              </Button>
              <Button 
                //to='/'
                color="inherit" 
                //className={clsx(classes.fontSize, classes.link)} 
                className={classes.fontSize}
                onClick={() => history.push('/')}
              >
                Home
              </Button>
              <Button 
                color="inherit" 
                //className={clsx(classes.fontSize, classes.link)}
                classNname={classes.fontSize}
              >
                Contact
              </Button>
              <Button 
                // to='signin'
                color="inherit" 
                //className={clsx(classes.fontSize, classes.link)} 
                className={classes.fontSize}
                onClick={() => history.push('/signin')}
              >
                Login
              </Button>
              <Button
                //to='/signup' 
                color="inherit" 
                //className={clsx(classes.fontSize, classes.link)}
                className={classes.fontSize}
                onClick={() => history.push('/signup')}
              > 
                Register
              </Button>
            </Typography>
          </Toolbar>
        </AppBar>
      </div>
    </div>
  );
}
