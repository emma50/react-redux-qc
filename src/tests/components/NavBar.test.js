import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from '../../components/navBar/NavBar';

test('create NavBar snapshot', () => {
  const component = renderer.create(
    <NavBar/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})