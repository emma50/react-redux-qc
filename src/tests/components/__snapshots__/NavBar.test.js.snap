// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create NavBar snapshot 1`] = `
<div>
  <div
    className="makeStyles-root-1"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-2 mui-fixed MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          className="MuiTypography-root makeStyles-title-4 MuiTypography-h6 MuiTypography-noWrap"
        >
          QUICK CREDIT
        </h6>
        <button
          aria-label="open drawer"
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          edge="end"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
      </div>
    </header>
    <div
      className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-6"
    >
      <div
        className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-7 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <div
          className="makeStyles-drawerHeader-8"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </span>
          </button>
        </div>
        <hr
          className="MuiDivider-root"
        />
        <ul
          className="MuiList-root MuiList-padding"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Apply Now
                </span>
              </div>
            </div>
          </a>
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Home
                </span>
              </div>
            </div>
          </a>
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Contact
                </span>
              </div>
            </div>
          </a>
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Login
                </span>
              </div>
            </div>
          </a>
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Register
                </span>
              </div>
            </div>
          </a>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="makeStyles-desktopRoot-11"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          className="MuiTypography-root makeStyles-desktopTitle-13 MuiTypography-h6"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "fontSize": "small",
                "fontWeight": "bold",
              }
            }
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              QUICK CREDIT
            </span>
          </button>
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root makeStyles-desktopTitle-13 makeStyles-fontWeight-15 MuiIconButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              Apply Now
            </span>
          </button>
          <button
            className="MuiButtonBase-root MuiIconButton-root makeStyles-fontSize-14 MuiIconButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              Home
            </span>
          </button>
          <button
            className="MuiButtonBase-root MuiIconButton-root makeStyles-fontSize-14 MuiIconButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              Contact
            </span>
          </button>
          <button
            className="MuiButtonBase-root MuiIconButton-root makeStyles-fontSize-14 MuiIconButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              Login
            </span>
          </button>
          <button
            className="MuiButtonBase-root MuiIconButton-root makeStyles-fontSize-14 MuiIconButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              Register
            </span>
          </button>
        </p>
      </div>
    </header>
  </div>
</div>
`;
